l10n_ch_sepa_credit_transfer/wizard/export_sepa.py

```diff
@@ -0,0 +1,303 @@
+        else:
+            raise orm.except_orm(
+                _('Error:'),
+                _("Payment Type Code '%s' is not supported. The only "
+                    "Payment Type Codes supported for SEPA Credit Transfers "
+                    "are 'pain.001.001.02', 'pain.001.001.03', "
+                    "'pain.001.001.04', 'pain.001.001.05',"
+                    " 'pain.001.003.03' and 'pain.001.001.03.ch.02'.") %
+                pain_flavor)
+
+        if pain_flavor == 'pain.001.001.03.ch.02':
+            module = 'l10n_ch_sepa_credit_transfer'
+
+        else:
+            module = 'account_banking_sepa_credit_transfer'
+        gen_args = {
+            'bic_xml_tag': bic_xml_tag,
+            'name_maxsize': name_maxsize,
+            'convert_to_ascii': convert_to_ascii,
+            'payment_method': 'TRF',
+            'pain_flavor': pain_flavor,
+            'sepa_export': sepa_export,
+            'file_obj': self.pool['banking.export.sepa'],
+            'pain_xsd_file': '%s/data/%s.xsd' % (module, pain_flavor)
+        }
+
+        pain_ns = {
+            'xs': 'http://www.w3.org/2001/XMLSchema-instance',
+            None: 'http://www.six-interbank-clearing.com/de/%s.xsd'
+                  % pain_flavor,
+        }
+
+        xml_root = etree.Element('Document', nsmap=pain_ns)
+        pain_root = etree.SubElement(xml_root, root_xml_tag)
+        pain_03_to_05 = [
+            'pain.001.001.03',
+            'pain.001.001.04',
+            'pain.001.001.05',
+            'pain.001.003.03'
+        ]
+
+        # A. Group header
+        group_header_1_0, nb_of_transactions_1_6, control_sum_1_7 = \
+            self.generate_group_header_block(
+                cr, uid, pain_root, gen_args, context=context)
+
+        transactions_count_1_6 = 0
+        total_amount = 0.0
+        amount_control_sum_1_7 = 0.0
+        lines_per_group = {}
+        # key = (requested_date, priority)
+        # values = list of lines as object
+        today = fields.date.context_today(self, cr, uid, context=context)
+        for payment_order in sepa_export.payment_order_ids:
+            total_amount = total_amount + payment_order.total
+            for line in payment_order.line_ids:
+                priority = line.priority
+                if payment_order.date_prefered == 'due':
+                    requested_date = line.ml_maturity_date or today
+                elif payment_order.date_prefered == 'fixed':
+                    requested_date = payment_order.date_scheduled or today
+                else:
+                    requested_date = today
+                key = (requested_date, priority)
+                if key in lines_per_group:
+                    lines_per_group[key].append(line)
+                else:
+                    lines_per_group[key] = [line]
+                # Write requested_date on 'Payment date' of the pay line
+                if requested_date != line.date:
```

Comment:
if we expect many lines here is better to collect a list of `(line_id, date)` to write everything at the end w/ a single query.
See [this example](https://github.com/camptocamp/bso_odoo/blob/master/odoo/local-src/sale_purchase_sourcing/wizards/wiz_sale_order_source.py#L285).